### ISSUE TITLE
Unify Orca Estimators evaluate method

### ISSUE THREAD

**Issue by yangw1234** [intel-analytics/analytics-zoo#3350]
## Current Issue
1. some estimator does not implement evaluate
2. some estimator does not support dataframe
3. the returned format might be different for different estimators
4. different estimators have different ways to specify metrics

## Proposal
1. all estimator should evaluate with xshard and dataframe
2. tf estiamtors should also support tf.data.Dataset and pytorch estimators should also support dataloaders
3. all estimators should return the evaluate result in the same format. Possibly a dict, like {"metrics_name": metric_value}
4. all estimators should have a similar API to specify an evaluation metric

## Tasks
- [x] implement evaluate in ray pytorch estimator
- [x] support dataframe in spark pytorch estimator
- [ ] support dataframe in bigdl estimator (tracked in another issue)
- [x] unify the returned format
- [x] unify API for specifying evaluation metric 
- [x] support specifying metrics in pytorch ray estimator #2994
